### PR TITLE
PR for SRVCOM-3248: Update the Serverless 1.34 release notes

### PR DESCRIPTION
**Pull request by kaldesai**
**Affected versions:** `serverless-docs-1.34`

**Tracking JIRA:** https://issues.redhat.com/browse/SRVCOM-3248

**Doc preview:** 

- [Serverless 1.34.0 release notes](https://82953--ocpdocs-pr.netlify.app/openshift-serverless/latest/about/serverless-release-notes.html#serverless-rn-1-34-0_serverless-release-notes)

- [Generally Available and Technology Preview features table](https://82953--ocpdocs-pr.netlify.app/openshift-serverless/latest/about/serverless-release-notes.html#serverless-tech-preview-features_serverless-release-notes)

- [Deprecated and removed features table](https://82953--ocpdocs-pr.netlify.app/openshift-serverless/latest/about/serverless-release-notes.html#serverless-deprecated-removed-features_serverless-release-notes)

### CODE REVIEW

File: modules/serverless-rn-1-34-0.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+* {ServerlessProductName} Serving transport encryption feature allows transporting data over secured and encrypted HTTPS connections using TLS. This is now available as a Technology Preview (TP) feature.
+
+* Go function using S2I builder is now available as a Technology Preview (TP) feature.
+
+* Multi-container support for Knative Serving allows you to use a single Knative service to deploy a multi-container pod. It also supports the the `readiness` and `liveness` probe values for multiple containers.
+
+* The new trigger filters feature is now available as a Technology Preview. These filters are enabled by default and allows users to specify a set of filter expressions, where each expression evaluates to either true or false for each event.
+
+* Autoscaling for Knative Kafka triggers is now enhanced with KEDA (Kubernetes Event-Driven Autoscaling). Autoscaling using CMA/LEDA further enhances performance by optimizing resource allocation for Kafka triggers, ensuring better scalability in event-driven workloads.
```
Comment (pierDipi) [inline modules/serverless-rn-1-34-0.adoc:37]:
```suggestion
* Autoscaling for Knative Kafka triggers is now enhanced with KEDA (Kubernetes Event-Driven Autoscaling). Autoscaling using CMA/KEDA further enhances performance by optimizing resource allocation for Kafka triggers and KafkaSources, ensuring better scalability in event-driven workloads.
```

File: modules/serverless-rn-1-34-0.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+* Knative Eventing now offers support for data in transit encryption (Eventing TLS) as a Technology Preview. You can configure Knative Eventing components to expose HTTPS addresses as well as add user-provided CA trust bundles to clients.
+
+[id="fixed-issues-1-34-0_{context}"]
+== Fixed issues
+
+* Previously, KafkaSource would incorrectly remain in the Ready status even when the `KafkaSource.spec.net.tls.key` failed to load. This issue has been resolved. An error is now reported when creating a Kafka Broker, KafkaChannel, KafkaSource, or KafkaSink with unsupported TLS certificates in PKCS #1 (Public-Key Cryptography Standards #1) format, ensuring proper handling and notification of configuration issues.
+
+* The controller incorrectly requeued the wrong object type (Namespace), causing 'resource not found' log errors. This issue is now resolved, and the controller now handles object requeuing, ensuring more accurate logging and resource management.
```
Comment (pierDipi) [inline modules/serverless-rn-1-34-0.adoc:46]:
```suggestion
* The Eventing controller incorrectly requeued the wrong object type (Namespace), causing 'resource not found' log errors. This issue is now resolved, and the controller now handles object requeuing, ensuring more accurate logging and resource management.
```

File: modules/serverless-rn-1-34-0.adoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+* {ServerlessProductName} 1.34 introduces a new feature that allows workflows to access different OpenShift clusters through configuration. This feature enables users to define REST calls within a workflow to seamlessly interact with multiple clusters.
+
+* The Jobs Service liveness checks is now enhanced to limit the time required to retrieve the leader status. A new system property, `kogito.jobs-service.management.leader-check.expiration-in-seconds`, has been introduced, allowing you to configure the maximum time allowed for the leader status check.
+
+* {ServerlessProductName} Serving transport encryption feature allows transporting data over secured and encrypted HTTPS connections using TLS. This is now available as a Technology Preview (TP) feature.
+
+* Go function using S2I builder is now available as a Technology Preview (TP) feature.
+
+* Multi-container support for Knative Serving allows you to use a single Knative service to deploy a multi-container pod. It also supports the the `readiness` and `liveness` probe values for multiple containers.
```
Comment (mgencur) [inline modules/serverless-rn-1-34-0.adoc:33]:
```suggestion
* Multi-container support for Knative Serving allows you to use a single Knative service to deploy a multi-container pod. It also supports the `readiness` and `liveness` probe values for multiple containers.
```

File: modules/serverless-rn-1-34-0.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+* The new trigger filters feature is now available as a Technology Preview. These filters are enabled by default and allows users to specify a set of filter expressions, where each expression evaluates to either true or false for each event.
```
Comment (mgencur) [inline modules/serverless-rn-1-34-0.adoc:35]: We already documented the new trigger filters for 1.32. No need to repeat this, IMO. It's not new.

File: modules/serverless-tech-preview-features.adoc [hunk 5]
```diff
@@ -13,40 +13,50 @@ The following table provides information about which {ServerlessProductName} fea
...
-|Feature |1.31|1.32|1.33
+|Feature |1.32|1.33|1.34
 
-|OpenShift Serverless Logic
+|Eventing TLS
+|-
+|-
 |TP
+
+|Encryption Serving
```
Comment (mgencur) [inline modules/serverless-tech-preview-features.adoc:23]: I suggest that we call this "Serving TLS" to align with "Eventing TLS". It's basically the same feature. Or we call both "Eventing Transport Encryption" and "Serving Transport Encryption".

Let's also remove the row "Serving TLS for internal traffic" from the "Deprecated and removed features tracker" table as it's confusing. We now have a new implementation of this TLS stuff as indicated above.

File: modules/serverless-rn-1-34-0.adoc (hunk 1 above)
Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc:37]: Done!

File: modules/serverless-rn-1-34-0.adoc (hunk 2 above)
Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc:46]: Done

File: modules/serverless-rn-1-34-0.adoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+:_content-type: REFERENCE
+[id="serverless-rn-1-34-0_{context}"]
+= Red Hat {ServerlessProductName} 1.34
+
+{ServerlessProductName} 1.34 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes:
+
+[id="new-features-1-34-0_{context}"]
+== New features
+
+* {ServerlessProductName} now uses Knative Serving 1.13.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc:14]: Fixed

File: modules/serverless-rn-1-34-0.adoc (hunk 3 above)
Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc:33]: Fixed

File: modules/serverless-rn-1-34-0.adoc (hunk 4 above)
Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc:35]: Okay, removed!

File: modules/serverless-tech-preview-features.adoc (hunk 5 above)
Comment (kaldesai) [reply modules/serverless-tech-preview-features.adoc:23]: Fixed and removed

**Comment by briandooley** [comment]
@nainaz Can we get your sign off on this please?

File: modules/serverless-rn-1-34-0.adoc [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+* {ServerlessProductName} now uses Kourier 1.14.
+* {ServerlessProductName} now uses Knative (`kn`) CLI 1.14.
+* {ServerlessProductName} now uses Knative for Apache Kafka 1.14.
+* The `kn func` CLI plugin now uses `func` 1.15.
+
+* It is now possible to support mutliple configuration for OpenAPI within the same namespace.
+
+* Encryption Serving is now available as a Technology Preview (TP) feature.
+
+* {ServerlessOperatorName} is now available for ARM-based clusters as a Tech Preview feature.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc:25]: Fixed!

File: modules/serverless-rn-1-34-0.adoc [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+* {ServerlessProductName} 1.34 introduces a new feature that allows workflows to access different OpenShift clusters through configuration. This feature enables users to define REST calls within a workflow to seamlessly interact with multiple clusters.
+
+* The Jobs Service liveness checks is now enhanced to limit the time required to retrieve the leader status. A new system property, `kogito.jobs-service.management.leader-check.expiration-in-seconds`, has been introduced, allowing you to configure the maximum time allowed for the leader status check.
+
+* {ServerlessProductName} Serving transport encryption feature allows transporting data over secured and encrypted HTTPS connections using TLS. This is now available as a Technology Preview (TP) feature.
+
+* Go function using S2I builder is now available as a Technology Preview (TP) feature.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc:33]: It is a singular one.

File: modules/serverless-rn-1-34-0.adoc [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+* {ServerlessProductName} Serving transport encryption feature allows transporting data over secured and encrypted HTTPS connections using TLS. This is now available as a Technology Preview (TP) feature.
+
+* Go functions using S2I builder are now available as a Technology Preview (TP) feature for Linux and Mac developers, allowing them to implement and build Go functions on these platforms.
+
+* Multi-container support for Knative Serving allows you to use a single Knative service to deploy a multi-container pod. It also supports the `readiness` and `liveness` probe values for multiple containers.
+
+* Autoscaling for Knative Kafka triggers is now enhanced with KEDA (Kubernetes Event-Driven Autoscaling) as a Technology Preview (TP). Autoscaling using CMA/LEDA further enhances performance by optimizing resource allocation for Kafka triggers and KafkaSources, ensuring better scalability in event-driven workloads.
```
Comment (mgencur) [inline modules/serverless-rn-1-34-0.adoc:37]: LEDA -> KEDA

Comment (kaldesai) [reply modules/serverless-rn-1-34-0.adoc]: Fixed!

File: modules/serverless-rn-1-34-0.adoc [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+* Knative Eventing now offers support for data in transit encryption (Eventing TLS) as a Technology Preview (TP) feature. You can configure Knative Eventing components to expose HTTPS addresses as well as add user-provided CA trust bundles to clients.
+
+[id="fixed-issues-1-34-0_{context}"]
+== Fixed issues
+
+* Previously, KafkaSource would incorrectly remain in the Ready status even when the `KafkaSource.spec.net.tls.key` failed to load. This issue has been resolved. An error is now reported when creating a Kafka Broker, KafkaChannel, KafkaSource, or KafkaSink with unsupported TLS certificates in PKCS #1 (Public-Key Cryptography Standards #1) format, ensuring proper handling and notification of configuration issues.
+
+* The Eventing controller incorrectly requeued the wrong object type (Namespace), causing 'resource not found' log errors. This issue is now resolved, and the controller now handles object requeuing, ensuring more accurate logging and resource management.
```
Comment (adellape) [inline modules/serverless-rn-1-34-0.adoc:46]:
```suggestion
* The Eventing controller incorrectly requeued the wrong object type (`Namespace`), causing "resource not found" log errors. This issue is now resolved, and the controller now handles object requeuing, ensuring more accurate logging and resource management.
```

File: modules/serverless-rn-1-34-0.adoc [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+* Autoscaling for Knative Kafka triggers is now enhanced with KEDA (Kubernetes Event-Driven Autoscaling) as a Technology Preview (TP). Autoscaling using CMA/KEDA further enhances performance by optimizing resource allocation for Kafka triggers and KafkaSources, ensuring better scalability in event-driven workloads.
+
+* Knative Eventing now offers support for data in transit encryption (Eventing TLS) as a Technology Preview (TP) feature. You can configure Knative Eventing components to expose HTTPS addresses as well as add user-provided CA trust bundles to clients.
+
+[id="fixed-issues-1-34-0_{context}"]
+== Fixed issues
+
+* Previously, KafkaSource would incorrectly remain in the Ready status even when the `KafkaSource.spec.net.tls.key` failed to load. This issue has been resolved. An error is now reported when creating a Kafka Broker, KafkaChannel, KafkaSource, or KafkaSink with unsupported TLS certificates in PKCS #1 (Public-Key Cryptography Standards #1) format, ensuring proper handling and notification of configuration issues.
```
Comment (adellape) [inline modules/serverless-rn-1-34-0.adoc:44]:
```suggestion
* Previously, `KafkaSource` objects would incorrectly remain in the `Ready` status even when the `KafkaSource.spec.net.tls.key` failed to load. This issue has been resolved. An error is now reported when creating a Kafka `Broker`, `KafkaChannel`, `KafkaSource`, or `KafkaSink` object with unsupported TLS certificates in PKCS #1 (Public-Key Cryptography Standards #1) format, ensuring proper handling and notification of configuration issues.
```

File: modules/serverless-rn-1-34-0.adoc [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+* {ServerlessProductName} now uses Knative for Apache Kafka 1.14.
+* The `kn func` CLI plugin now uses `func` 1.15.
+
+* It is now possible to support multiple configuration for OpenAPI within the same namespace.
+
+* Encryption Serving is now available as a Technology Preview (TP) feature.
+
+* {ServerlessOperatorName} is now available for ARM-based clusters as a Technology Preview (TP) feature.
+
+* {ServerlessProductName} 1.34 introduces a new feature that allows workflows to access different OpenShift clusters through configuration. This feature enables users to define REST calls within a workflow to seamlessly interact with multiple clusters.
```
Comment (adellape) [inline modules/serverless-rn-1-34-0.adoc:27]:
```suggestion
* {ServerlessProductName} 1.34 introduces a new feature that allows workflows to access different {ocp-product-title} clusters through configuration. This feature enables users to define REST calls within a workflow to seamlessly interact with multiple clusters.
```

File: modules/serverless-rn-1-34-0.adoc [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+* {ServerlessProductName} Serving transport encryption feature allows transporting data over secured and encrypted HTTPS connections using TLS. This is now available as a Technology Preview (TP) feature.
+
+* Go functions using S2I builder are now available as a Technology Preview (TP) feature for Linux and Mac developers, allowing them to implement and build Go functions on these platforms.
+
+* Multi-container support for Knative Serving allows you to use a single Knative service to deploy a multi-container pod. It also supports the `readiness` and `liveness` probe values for multiple containers.
+
+* Autoscaling for Knative Kafka triggers is now enhanced with KEDA (Kubernetes Event-Driven Autoscaling) as a Technology Preview (TP). Autoscaling using CMA/KEDA further enhances performance by optimizing resource allocation for Kafka triggers and KafkaSources, ensuring better scalability in event-driven workloads.
```
Comment (adellape) [inline modules/serverless-rn-1-34-0.adoc:37]:
```suggestion
* Autoscaling for Knative Kafka triggers is now enhanced with KEDA (Kubernetes Event-Driven Autoscaling) as a Technology Preview (TP). Autoscaling using CMA/KEDA further enhances performance by optimizing resource allocation for Kafka triggers and `KafkaSource` objects, ensuring better scalability in event-driven workloads.
```

**Comment by kaldesai** [comment]
> LGTM from QE

Thank you so much for reviewing the release notes.

**Review by adellape — APPROVED** [review]

**Comment by nainaz** [comment]
Done on Oct 10

**Pull request merged**